src/util/url-filter.js

```diff
@@ -0,0 +1,23 @@
+//Remove unnecessary query params from the URL to avoid redundancy
+const NORMALIZE_OPTS = {
+    'stripWWW': false,
+    'removeDirectoryIndex': false,
+    'removeTrailingSlash': false,
+    'stripFragment': true,
+    'normalizeProtocol': false
+}
+const GSEARCH_SUBSTR = 'www.google'
+const FB_SUBSTR = 'www.facebook'
+const normalizeUrl = require("normalize-url")
+export default function urlFilter(url) {
+    const parts = url.split('.')
```

Comment:
The main purpose of this part is to get the domain which can be used to make a decision on how it should be normalised? If so, stay away from `String.prototype.split` and use the [`URL` WebAPI](https://developer.mozilla.org/en-US/docs/Web/API/URL) to try and extract these parts.

Do note that you will want to normalize the passed-in URL (via `normalizeUrl` package) before doing anything else, to ensure compatibility with different types of formatted URLs. Getting rid of parts that do not have meaning in this function's context (like `www`, hash fragment).

Reply:
I've noticed that `normalize-url` can jumble up the parameters so I wouldn't be able to extract the desirable parts for say google search URLs

Reply:
Can you give an example of one of these? Not clear on what "jumble up" means here

Reply:
You could also use the native [`URLSearchParams` interface](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) to easily mutate the search params without string manipulation. You can get the `searchParams` property if you parse a URL using the `URL` API.

An example script using that API that should remove every param apart from those defined in `rules` Set, which you can run in browser console:
```js
let url = 'https://www.google.com.vn/search?ei=mb_5WcW7H8qs0AShu4ywDg&q=yo+waddup+query&oq=yo+waddup+query'
let rules = new Set(['q'])

let parsedUrl = new URL(url)

for (const param of parsedUrl.searchParams.keys()) {
    if (!rules.has(param)) {
        parsedUrl.searchParams.delete(param)
    }
}

console.log('before normalizing', url)
console.log('after normalizing', parsedUrl.toString())

```

Reply:
Example of parameters getting reordered https://runkit.com/embed/7tx49vfo2q3n

Reply:
Yeah, no problems then if it's just reordering; order isn't important here.